pyne/serpent.py

```diff
@@ -370,3 +370,164 @@ def parse_det(detfile, write_py=False):
     exec(f, {}, det)
 
     return det
+
+def parse_coe(coefile, grabB1=True):
+    """ Used to parse output from serpent 2's ability to branch problem variables
+    in order to generate group constants at several states. 
+    
+    See the serpent wiki at:
+    http://serpent.vtt.fi/mediawiki/index.php/Automated_burnup_sequence
+
+    Example:
+        say that you apply this to a serpent output with fuel temperature changes
+        named fuel0 through fuel14. Then, control rod positions were altered with
+        branches rod0 through rod4. In total, this would make 75 sets of different
+        group constants since all combinations are covered. To access INF_KINF with
+        (ie k_infinity with no B1 corrections) in universe 2
+        fuel0, rod0 at day 0 of burnup, you'd use:
+
+        coe[1]         ["2"]     ["fuel0"]["rod0"]["INF_KINF"]
+        #   ^ dep step  ^ uni ID  ^branches       ^ parameter name
+        
+        all parameters serpent gives are available by default
+
+    Parameters
+    ----------
+    coefile : str or file-like object
+        Path to *.coe file or a *.coe file handle.
+
+    grabB1 : bool
+        whether to include results from the B1 corrected spectrum
+        if group constants were generated on full-core, set this to false
+        to save memory.
+
+    Returns
+    -------
+    coe : dict
+        Nested dictionary of the parsed coe file. First dimension key is burnup index (1-indexed). Next
+        is the universe group constants were generated in. Then, branch names represent
+        the keys, in the order that serpent printed them. On the final dimension, the parameters returned by serpent
+        make up the remaining keys. 
+
+        note: B1 is used for leakage-corrected group constants generated on an infinite
+        lattice. Don't use these values otherwise.
+
+    """
+    if isinstance(coefile, basestring):
+        f = open(coefile, 'r')
+    else:
+        f = coefile
```

Comment:
f is never closed explicitly here.  Also, it is generally inadvisable to leave a file handle open longer than it needs to be.  You should be using a context manager (with-statement) for this.  This will guarantee that the file is always appropriately closed.